File: D.S.A_python/linear_search.py
#linear search
import random
n = int(input("enter the size: "))
arr = [0]*n
for i in range(n):
    arr[i] = random.randint(-10,10)
print("the array:")
for i in range(n):
    print(arr[i],end=" ")

x = int(input("\nenter the element you want: "))
result = -1

for i in range(n):
    if arr[i]==x:
        result = i
        break

if result!=-1:
    print("the index is: ",result)
else:
    print("unna elemnts enter chei ra pooka!!!")

File: D.S.A_c/linear_search.c
#include<stdio.h>
int main(){
    int n;int search;
    scanf("%d",&n);
    int arr[n];
    for(int i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }
    printf("Enter element to search: ");
    scanf("%d",&search);
    for(int j=0;j<n;j++){
        if(arr[j]==search){
            printf("Element found at index %d\n",j);
            return 0;
        }
    }
    printf("Element not found!");
    return 0;
}